scheduler/replica_scheduler.go

```diff
@@ -84,39 +83,46 @@ func (rcs *ReplicaScheduler) ScheduleReplica(replica *longhorn.Replica, replicas
 		nodeDisksMap[node.Name] = disks
 	}
 
-	diskCandidates := rcs.getDiskCandidates(nodeCandidates, nodeDisksMap, replicas, volume, true)
+	diskCandidates, verboseErrors := rcs.getDiskCandidates(nodeCandidates, nodeDisksMap, replicas, volume, true)
 
 	// there's no disk that fit for current replica
 	if len(diskCandidates) == 0 {
 		logrus.Errorf("There's no available disk for replica %v, size %v", replica.ObjectMeta.Name, replica.Spec.VolumeSize)
-		return nil, nil
+		return nil, verboseErrors, nil
 	}
 
 	// schedule replica to disk
 	rcs.scheduleReplicaToDisk(replica, diskCandidates)
 
-	return replica, nil
+	return replica, nil, nil
 }
 
-func (rcs *ReplicaScheduler) getNodeCandidates(nodesInfo map[string]*longhorn.Node, schedulingReplica *longhorn.Replica) map[string]*longhorn.Node {
+func (rcs *ReplicaScheduler) getNodeCandidates(nodesInfo map[string]*longhorn.Node, schedulingReplica *longhorn.Replica) (nodeCandidates map[string]*longhorn.Node, verboseErrors util.VerboseErrors) {
 	if schedulingReplica.Spec.HardNodeAffinity != "" {
 		node, exist := nodesInfo[schedulingReplica.Spec.HardNodeAffinity]
```

Comment:
Can we use string formatting w/ interpolation to add `schedulingReplica.Spec.HardNodeAffinity` into the error message for having much detailed info? like fmt.Sprintf(longhorn.SchedulerErrorHardNodeAffinityNotSatisfied, schedulingReplica.Spec.HardNodeAffinity).